wgpu/src/api/render_pipeline.rs

```diff
@@ -103,32 +123,53 @@ pub struct FragmentState<'a> {
     /// Advanced options for when this pipeline is compiled
     ///
     /// This implements `Default`, and for most users can be set to `Default::default()`
+    #[builder(default)]
     pub compilation_options: PipelineCompilationOptions<'a>,
     /// The color state of the render targets.
     pub targets: &'a [Option<ColorTargetState>],
 }
 #[cfg(send_sync)]
 static_assertions::assert_impl_all!(FragmentState<'_>: Send, Sync);
 
+impl<'a> FragmentState<'a> {
+    #[allow(missing_docs)]
+    pub fn from_module(
+        module: &'a ShaderModule,
+    ) -> FragmentStateBuilder<
+        'a,
+        (
+            bon::private::Set<&'a ShaderModule>,
+            bon::private::Unset<bon::private::Optional>,
+            bon::private::Unset<bon::private::Optional>,
+            bon::private::Unset<bon::private::Required>,
+        ),
+    > {
+        Self::builder().module(module)
+    }
+}
+
 /// Describes a render (graphics) pipeline.
 ///
 /// For use with [`Device::create_render_pipeline`].
 ///
 /// Corresponds to [WebGPU `GPURenderPipelineDescriptor`](
 /// https://gpuweb.github.io/gpuweb/#dictdef-gpurenderpipelinedescriptor).
-#[derive(Clone, Debug)]
+#[derive(bon::Builder, Clone, Debug)]
 pub struct RenderPipelineDescriptor<'a> {
     /// Debug label of the pipeline. This will show up in graphics debuggers for easy identification.
+    #[builder(default, into)]
```

Comment:
Given that `Label` is just an `Option<&str>`, I'd suggest to inline that type alias so that `bon` recognizes it as an `Option` under the hood.

For example with the current 

```rust
#[builder(default, into)] label: Label<'a>
```

the following setters will be generated:

```rust
fn label(self, value: impl Into<Option<&str>>) -> NextBuilderState
fn maybe_label(self, value: Option<impl Into<Option<&str>>) -> NextBuilderState
```

which gives unnecessary `Option` nesting. If you simplify it to this:

```rust
label: Option<&'a str>
```
then the following setters will be generated:

```rust
fn label(self, value: &str) -> NextBuilderState
fn maybe_label(self, value: Option<&str>) -> NextBuilderState
```

which removes the `Option` nesting, and greatly simplifies the code